templates/templates.json

```diff
@@ -0,0 +1,117 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+
+    "definitions": {
+        "Test_run": {
+            "type": "object",
+	    "Properties": {
+                "uuid": {"type": "string"},
+		"start_time": {"type": "string"},
+		"end_time": {"type": "string"},
+		"build": {"type": "string"},
+		"status": {
+		    "type": "array",
+		    "items": {"$ref": "#/definitions/Status"},
+		    "default": []
+		},
+		"test_cases": {
+		    "type": "array",
+		    "items": {"type":"string"},
+		    "default": []
+		},
+		"test_suites": {
+		    "type": "array",
+		    "items": {"type":"string"},
+		    "default": []
+		}
+	    }
+	},
+	"Test_suite": {
+            "type": "object",
+	    "Properties": {
+                "uuid": {"type": "string"},
+		"test_cases": {
+		    "type": "array",
+		    "items": {"type": "string"},
+		    "default": []
+		},
+		"keywords": {
+		    "type": "array",
+		    "items": {"type": "string"},
+		    "default": []
+		}
+	    }
+	},
+	"Test_case": {
+            "type": "object",
+	    "Properties": {
+                "uuid": "string",
+		"title": "string",
+		"steps": {
+		    "type": "array",
+		    "items": {"$ref": "#/definitions/Step/"},
+		    "default": []
+		},
+		"pass_cond": "string"
+	    }
+	},
+	"Step": {
+            "type": "object",
+	    "Properties": {
+		"uuid": "string",
+		"title": "string",
+		"action": "string",
+		"parameter": {
+                    "type": "array",
```

Comment:
My understanding of how the arrays work is the second option, ["-p public", "-s starttime", "-b firefox"], for example.

I'm currently a bit hesitant as to how we want to list the parameters. I thought about it and seriously considered just keeping all parameters as one long string and using the argparse module to extract them.

Part of the reasoning here was that parameters can be provided as flags (-v or --verbose), key value pairs (-b firefox), or flag combinations (-zcvf). I figured the string array gave us some basic organization while maintaining a fair amount of flexibility.

@ty-desj let me know what you think.

Reply:
Ahh, I think I misunderstood the use of this parameter object. I was thinking about parameters for the action. For example with 'click' would require a target to click, or with 'type' needing a target and keys to input.

Reply:
Yeah, it can be that too. While python functions don't natively support things like flags, we might want to be able to support parameters in those formats depending on how our clients are used to working (or what the tools we integrate with use).

Either way, I wanted to have something that intuitively supported both positional arguments and named arguments alongside the others listed in my previous comment.